commons/src/main/java/life/qbic/application/commons/Result.java

```diff
@@ -0,0 +1,242 @@
+    Objects.requireNonNull(value);
+    return new Result<>(value);
+  }
+
+  /**
+   * Static constructor method for creating a result object instance of type <code>V,E</code>
+   * wrapping an exception <code>E</code>.
+   *
+   * @param exception the exception to get wrapped in a result object
+   * @return a new result object instance
+   */
+  public static <V, E extends Exception> Result<V, E> failure(E exception) {
+    Objects.requireNonNull(exception);
+    return new Result<>(exception);
+  }
+
+  private Result(V value) {
+    this.value = value;
+    this.exception = null;
+    this.type = Type.SUCCESS;
+  }
+
+  private Result(E exception) {
+    this.value = null;
+    this.exception = exception;
+    this.type = Type.FAILURE;
+  }
+
+  /**
+   * Access the wrapped value if present
+   *
+   * @return the wrapped value
+   * @throws NoSuchElementException if no value exists in the result object
+   */
+  V value() throws NoSuchElementException {
+    if (Objects.isNull(value)) {
+      throw new NoSuchElementException("Result with exception has no value.");
+    }
+    return value;
+  }
+
+  /**
+   * Access the wrapped exception if present
+   *
+   * @return the wrapped exception
+   * @throws NoSuchElementException if no error exists in the result object
+   */
+  E exception() throws NoSuchElementException {
+    if (Objects.isNull(exception)) {
+      throw new NoSuchElementException("Result with value has no exception.");
+    }
+    return exception;
+  }
+
+  /**
+   * Returns <code>true</code>, if the result object contains an error. Is always the negation of
+   * {@link Result#isSuccess()} ()}.
+   * <p>So <code>{@link Result#isFailure()} ()} == !{@link Result#isSuccess()}</code></p>
+   *
+   * @return true, if the result object has an error, else false
+   */
+  public Boolean isFailure() {
+    return type.equals(Type.FAILURE);
+  }
+
+  /**
+   * Returns <code>true</code>, if the result object contains an error. Is always the negation of
+   * {@link Result#isFailure()} ()}.
+   * <p>So <code>{@link Result#isSuccess()} ()} == !{@link Result#isFailure()} ()}</code></p>
+   *
+   * @return true, if the result object has a value, else false
+   */
+  public Boolean isSuccess() {
+    return type.equals(Type.SUCCESS);
+  }
+
+  /**
+   * <p>Maps the current result object to a consumer function, that expects the same input
+   * type <code>V</code> as the result's value type and produces a result object of type
+   * <code>U,E</code>.</p>
+   *
+   * <p>If the current result contains an error (and therefore has no value), the created result
+   * object will contain the error of type <code>E</code> of the input result object.
+   * </p>
+   *
+   * @param function a function transforming data of type <code>V</code> to <code>U</code>
+   * @return a new result object instance of type <code>U,E</code>
+   */
+  public <U> Result<U, ? extends E> map(Function<V, U> function) {
```

Comment:
```suggestion
  public <U> Result<U, ? extends Exception> map(Function<V, U> function) {
```
This needs to extend Exception itself. If you have a Result<Integer,IllegalArgumentException> res and you fail to do res.map(it -> it.fooBar()) then you get a Result<Integer,NoSuchOperationException> or similar. These exceptions do not have any relationship except that they are exceptions